### PR TITLE
Add support for commit cross references

### LINKED ISSUE #22628: Allow linking to commits in other repos
### Feature Description

GitHub allows referring to issues in other repos using `owner/repo#issue` (which we support already, just for clarity).
Similarly, it allows referencing commits in other repos using `owner/repo@sha`, i.e. delvh/gitea@51a92cb8218b6702a5a0c8f921eda02456332748 (which Gitea doesn't).
Gitea should support it too.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by KN4CK3R**
Fixes #22628

This PR adds cross references for commits by using the format `owner/repo@commit` . References are rendered like [go-gitea/lgtm@6fe88302](#dummy).

### CODE REVIEW

**Comment by yardenshoham** [comment]
I think it'd be really cool if the commit sha would be in a code span like in GitHub:
go-gitea/lgtm@6fe8830276a06b01a2f8782f55d9c05d6fc54b47

Markdown source:
```
go-gitea/lgtm@6fe8830276a06b01a2f8782f55d9c05d6fc54b47
```

**Review by yardenshoham — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by KN4CK3R** [comment]
I saw that too but what's the benefit of the code span?

**Comment by yardenshoham** [comment]
Looks cute

File: modules/markup/html.go [hunk 1]
```diff
@@ -907,6 +911,23 @@ func issueIndexPatternProcessor(ctx *RenderContext, node *html.Node) {
...
+	next := node.NextSibling
+
+	for node != nil && node != next {
+		found, ref := references.FindRenderizableCommitCrossReference(node.Data)
+		if !found {
+			return
+		}
+
+		reftext := ref.Owner + "/" + ref.Name + "@" + base.ShortSha(ref.CommitSha)
+		link := createLink(util.URLJoin(setting.AppURL, ref.Owner, ref.Name, "commit", ref.CommitSha), reftext, "commit")
```
Comment (delvh) [inline modules/markup/html.go]: Absolute link? Aren't we trying at the moment to replace them with relative links?

**Comment by delvh** [comment]
Actually, I don't really understand why only the commit should get a code-span, and not the whole link…

File: modules/markup/html.go (hunk 1 above)
Comment (lunny) [reply modules/markup/html.go]: Use `setting.AppSubURL` instead.

**Comment by KN4CK3R** [comment]
Then lets have a vote:

🚀 [go-gitea/gitea@abcd1234](#dummy)
👍  [go-gitea/gitea@`abcd1234`](#dummy)
😄 [`go-gitea/gitea@abcd1234`](#dummy)

**Review by lunny — APPROVED** [review]

**Comment by jolheiser** [comment]
Honestly I would almost prefer:
[go-gitea/gitea`@abcd1234`](https://github.com/go-gitea/gitea/pull/22645#dummy)
but _also_ changing the issue/pr one to [go-gitea/gitea`#1234`](https://github.com/go-gitea/gitea/pull/22645#dummy)
I think it makes the important part more obvious imo

But with the vote at hand, I just vote for consistency.

**Pull request merged**